### PR TITLE
fira-math: init at 0.3.4

### PR DESCRIPTION
**Pull request by loicreynier**
###### Description of changes

Add [Fira Math](https://github.com/firamath/firamath), a Fira Sans extension with Unicode math support.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/fira-math/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "fira-math";
+  version = "0.3.4";
+  user = "firamath";
+  repo = "firamath";
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchurl {
+    url = "https://github.com/${user}/${repo}/releases/download/v${version}/FiraMath-Regular.otf";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/fira-math/default.nix:13]:
```suggestion
in
stdenv.mkDerivation {
  inherit pname version;

  src = fetchurl {
    url = "https://github.com/firamath/firamath/releases/download/v${version}/FiraMath-Regular.otf";
```

File: pkgs/data/fonts/fira-math/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  dontUnpack = true;
+
+  installPhase = ''
+    install -D $src -t $out/share/fonts/opentype/
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/${user}/${repo}";
+    description = "Math font with Unicode math support based on FiraSans and FiraGO";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/fira-math/default.nix:26]:
```suggestion
    description = "Math font with Unicode math support based on FiraSans and FiraGO";
    homepage = "https://github.com/firamath/firamath";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Please rebase on master

File: pkgs/data/fonts/fira-math/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    url = "https://github.com/firamath/firamath/releases/download/v${version}/FiraMath-Regular.otf";
+    sha256 = "sha256-ICjL091NjAzxYIUg60dZlWqDpnkx17bY58MTUgGG41s=";
+    name = "FiraMath-Regluar.otf";
+  };
+
+  dontUnpack = true;
+
+  installPhase = ''
+    install -D $src -t $out/share/fonts/opentype/
+  '';
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/fira-math/default.nix:20]: We can do this in only fetchurl and postFetch. see https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/fira/default.nix

Comment (loicreynier) [reply pkgs/data/fonts/fira-math/default.nix:20]:
```nix
{ lib, fetchurl, stdenv }:

let
  version = "0.3.4";
in
fetchurl {
  name = "fira-math-${version}";

  url = "https://github.com/firamath/firamath/releases/download/v${version}/FiraMath-Regular.otf";

  postFetch = ''
    install -D $downloadedFile -t $out/share/fonts/opentype/
  '';

  sha256 = "sha256-ICjL091NjAzxYIUg60dZlWqDpnkx17bY58MTUgGG41s=";

  meta = with lib; {
    description = "Math font with Unicode math support based on FiraSans and FiraGO";
    homepage = "https://github.com/firamath/firamath";
    license = licenses.ofl;
    maintainers = [ maintainers.loicreynier ];
    platforms = platforms.all;
  };
}
```

While trying to do so I have an error at the `$out` directory creation that I can't solve.

Comment (SuperSandro2000) [reply pkgs/data/fonts/fira-math/default.nix:20]: Looks correct to me. What is the error exactly?

Comment (loicreynier) [reply pkgs/data/fonts/fira-math/default.nix:20]:
```
install: failed to access '/nix/store/9pq9bzkf22j889j41dhpb8rk3p7d88v5-fira-math-0.3.4/share/fonts/opentype/': Not a directory
```
or with `mkdir`

```
mkdir: cannot create directory '/nix/store/9pq9bzkf22j889j41dhpb8rk3p7d88v5-fira-math-0.3.4': File exists
```

**Pull request closed without merging**